modules/oe_theme_helper/src/Plugin/Field/FieldFormatter/FeaturedMediaRenderedEntityFormatter.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_theme_helper\Plugin\Field\FieldFormatter;
+
+use Drupal\Core\Field\FieldItemInterface;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Field\Plugin\Field\FieldFormatter\EntityReferenceFormatterBase;
+use Drupal\file\Entity\File;
+
+/**
+ * Plugin implementation of the 'Featured media as entity' formatter.
+ *
+ * @FieldFormatter(
+ *   id = "oe_featured_media_entity_view",
+ *   label = @Translation("Rendered entity"),
+ *   description = @Translation("Display the entity of the referenced media and the caption."),
+ *   field_types = {
+ *     "oe_featured_media"
+ *   }
+ * )
+ */
+class FeaturedMediaRenderedEntityFormatter extends EntityReferenceFormatterBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode) {
+    $elements = [];
+
+    foreach ($items as $delta => $item) {
+      $elements[$delta] = $this->viewElement($item);
+    }
+
+    return $elements;
+  }
+
+  /**
+   * Renders a single field item.
+   *
+   * @param \Drupal\Core\Field\FieldItemInterface $item
+   *   The individual field item.
+   *
+   * @return array
+   *   The ECL media-container parameters.
+   */
+  protected function viewElement(FieldItemInterface $item): array {
+    $element = [];
+    $element['_front_params']['description'] = $item->caption;
+
+    switch ($item->entity->bundle()) {
+      case 'remote_video':
+        $entity = $item->entity;
+        $element['_front_params']['embedded_media'] = $entity->get('oe_media_oembed_video')->view([
+          'type' => 'oembed',
+          'label' => 'hidden',
+        ]);
+        break;
+
+      case 'image':
+        $file = File::load($item->target_id);
```

Comment:
Need to ensure that file entity exists. See https://github.com/openeuropa/oe_theme/pull/513/files#diff-175fc9d25ec415590a71139896c6b64dR38

Reply:
done

Reply:
actually here it does not make sense to check for FileInterface, I instantiate a \Drupal\file\Entity\File so thats clear

Reply:
File entity can be removed, in this case `$file->url()` will throw an error